akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2PersistentClientSpec.scala

```diff
@@ -182,6 +184,62 @@ class Http2PersistentClientSpec extends AkkaSpecWithMaterializer(
           // request should have come in through another connection
           serverRequest2.clientPort should not be (clientPort)
         }
+        "when the first connection fails to materialize" inAssertAllStagesStopped new TestSetup(tls) {
+          var first = true
+          override def clientSettings = super.clientSettings.withTransport(ClientTransport.withCustomResolver((host, port) => {
+            if (first) {
+              first = false
+              // First request returns an address where we are not listening::
+              Future.successful(new InetSocketAddress("example.invalid", 80))
+            } else
+              Future.successful(server.binding.localAddress)
+          }))
+
+          client.sendRequest(
+            HttpRequest(
+              method = HttpMethods.POST,
+              entity = "ping",
+              headers = headers.`Accept-Encoding`(HttpEncodings.gzip) :: Nil
+            )
+              .addAttribute(requestIdAttr, RequestId("request-1"))
+          )
+          // need some demand on response side, otherwise, no requests will be pulled in
+          client.responsesIn.request(1)
+          client.requestsOut.ensureSubscription()
+
+          val serverRequest = server.expectRequest()
+          serverRequest.request.attribute(Http2.streamId) should not be empty
+          serverRequest.request.method shouldBe HttpMethods.POST
+          serverRequest.request.header[headers.`Accept-Encoding`] should not be empty
+          serverRequest.entityAsString shouldBe "ping"
+
+          // now respond
+          server.sendResponseFor(serverRequest, HttpResponse(entity = "pong"))
+
+          val response = client.expectResponse()
+          Unmarshal(response.entity).to[String].futureValue shouldBe "pong"
+          response.attribute(requestIdAttr).get.id shouldBe "request-1"
+        }
+      }
+      "eventually fail" should {
+        "when connecting keeps failing" inAssertAllStagesStopped new TestSetup(tls) {
```

Comment:
nitpick: it'd be great if the setting `akka.http.client.http2.max-persistent-attempts = 5` was set for this test instead of being global.